### DISCUSSION TITLE
About air playing 4K video.

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#1246] [Q&A]
When I play a video, the resolution is 4K, and the resolution of some videos is 5760 * 2160. I use stagevideo instead of video, but the video is still stuck.

I wonder if there is a better way to play 4K video smoothly, or if there are any special requirements for video format? Now I am using MP4 format video, and everything is normal when playing with the windows built-in player.

I develop desktop applications

**Comment by itlancer** [comment]
I also suffering with a lot of issues when trying to playback 4K (or higher) resolution videos.
`Video` class performance awful. `StageVideo` and `VideoTexture` better but have some limitations. And any of these three AIR approaches have worse performance comparing with "system" media players (for any OS), ffmpeg, VLC and others.

- Videos with resolution higher than 4K cannot playback https://github.com/airsdk/Adobe-Runtime-Support/issues/89
- High resolution (above 4K) textures not supported https://github.com/airsdk/Adobe-Runtime-Support/issues/196
- Some MP4 H.264 profiles cannot be playback properly (cause hang/crash) https://github.com/airsdk/Adobe-Runtime-Support/issues/139
- Some videos on render contains green strip when playback https://github.com/airsdk/Adobe-Runtime-Support/issues/88
- "waiting for the GPU" performance issues that cause lags https://github.com/airsdk/Adobe-Runtime-Support/issues/85 https://github.com/airsdk/Adobe-Runtime-Support/issues/307
- Also with latests AIR version I noticed excessive memory consumption when video playback https://github.com/airsdk/Adobe-Runtime-Support/issues/646

**Comment by Jason0247** [comment]
I have encountered all the above problems, and I really haven't found a good solution.

I have tried to use f4v format video. The effect is relatively better, but the compression quality is also required. In theory, it still can't reach 4K video playback.
